grails-app/assets/javascripts/document.js

```diff
@@ -401,22 +437,300 @@ function findDocumentById(documents, id) {
     return null;
 }
 
-var DocModel = function (doc) {
+var DocModel = function (doc, settings) {
```

Comment:
I am just wondering why code had duplicated from document.js's DocumentViewModel? Could DocumentViewModel be reused here?